core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -122,9 +232,11 @@ class DefaultAutoTopicCreationManager(
       override def onComplete(response: ClientResponse): Unit = {
         clearInflightRequests(creatableTopics)
         if (response.authenticationException() != null) {
-          warn(s"Auto topic creation failed for ${creatableTopics.keys} with authentication exception")
+          val authException = response.authenticationException()
```

Comment:
nit: lets not change this function at all, it's not related to what we are doing.